specs/SBOM_SPEC.md

```diff
@@ -0,0 +1,101 @@
+
+Multiple SBOMs may be "attached" to one image.
+Each SBOM may refer to the entire image, or to a specific part of that image.
+Exactly what the SBOM refers to is called the "scope" of that SBOM.
+
+SBOMs stored in an OCI registry are generally assumed to refer to other objects stored in that same registry.
+This document does not specify how these "links" are created.
+A naming-convention or the in-progress OCI `references` API are viable options.
+
+## Overall Layout
+
+An SBOM object is represented an [OCI Image Manifest V1](https://github.com/opencontainers/image-spec/blob/master/manifest.md).
+
+Each individual SBOM is represented as a `layer`, using a standard `descriptor`.
+The `layers` list is ordered, but no order is assumed or important for the SBOM documents.
+
+Here is an example manifest containing one SBOM, in the [SPDX](https://spdx.org) format:
+
+```json
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.oci.image.config.v1+json",
+    "size": 233,
+    "digest": "sha256:83bd5fb5b39f65f28e50a86d48fa79c07880befc292d92eebdc18531054b070c"
+  },
+  "layers": [
+    {
+      "mediaType": "text/spdx",
+      "size": 246,
+      "digest": "sha256:ed3ad03d3b87843b5419d7dce9d50a3e0f45554b2ba93bf378611cae6b450cff",
+    }
+  ]
+}
+```
+
+Multiple SBOMs may be attached, using multiple formats.
+This example shows two SBOMs, one in the SPDX format and one in the CycloneDX format:
+
+```json
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.oci.image.config.v1+json",
+    "size": 233,
+    "digest": "sha256:83bd5fb5b39f65f28e50a86d48fa79c07880befc292d92eebdc18531054b070c"
+  },
+  "layers": [
+    {
+      "mediaType": "text/spdx",
+      "size": 246,
+      "digest": "sha256:ed3ad03d3b87843b5419d7dce9d50a3e0f45554b2ba93bf378611cae6b450cff",
+    },
+    {
+      "mediaType": "application/vnd.cyclonedx",
+      "size": 462,
+      "digest": "sha256:e0851a4aa13657fc8dcd01e0e5e08cb817123ccb82e2c604b34f9ec9c1755e3f",
+    }
+  ]
+}
+```
+
+## MediaTypes
+
+The two main SBOM formats in use are SPDX and CyloneDX.
+The `mediaTypes` for these should be indicated in the `descriptor` for each `layer`.
+
+The `mediaTypes` are:
+
+* `application/vnd.cyclonedx`
+* `text/spdx`
+
+## Scopes
+
+SBOMs may refer to an entire object, or to a specific part of that object.
+This is called the `scope` of the SBOM.
+
+The `scope` should be indicated via an annotation on the `Descriptor`.
+The annotation key for this `scope` is: `dev.sigstore.sbom/scope`
+
+
+A descriptor with no scope is assumed to refer to the entire object.
+This is the same as the scope of `all`.
+
+Well-known scopes include:
+
+* `all`: the SBOM refers to the entire object.
+* `layers=N`: the SBOM refers to the Nth layer.
+* `layers=N..M`: the SBOM refers to the layer range from N to M, exclusive of M.
+* `path=<foo>`: the SBOM refers to file at path `foo` in the flattened image.
```

Comment:
Should this spec cover what implementations should do when confronted with two overlapping SBOMs?

E.g., if there's an SBOM for `all` and for layer `layers=N`, is the full bill of materials the union of the two? Does `layers=N` supercede the `all` in some way? Is that configuration to be considered invalid, and clients should error out?

Reply:
Added this to the top. IMO it should be up to clients. The SBOM landscape is too ambiguous right now to prescribe what should happen here for multiple SBOMs or overlapping ones.